src/hmem_cuda.c

```diff
@@ -754,11 +726,33 @@ int cuda_host_unregister(void *ptr)
 
 enum cuda_xfer_setting cuda_get_xfer_setting(void)
 {
+	enum cuda_xfer_setting hmem_cuda_xfer_setting;
+	int ret, param_value = 1;
+
+	ret = fi_param_get_bool(NULL, "hmem_cuda_enable_xfer", &param_value);
+	if (ret == -EINVAL){
+		fprintf(stderr, "FI_HMEM_CUDA_ENABLE_XFER was set to an invalid value by the user."
+			" FI_HMEM_CUDA_ENABLE_XFER is boolean and can be set to only 0 or 1. \n");
```

Comment:
We can probably afford to be a little less verbose here. Pedantically, a boolean parameter can be something other than 0/1 - e.g. `yes`/`no`, `on`/`off`, etc.